packages/cli/src/vercel-snapshot.ts

```diff
@@ -0,0 +1,53 @@
+import path from 'path';
+import { globby } from 'globby';
+import fsExtra from 'fs-extra';
+
+/**
+ * When deploying a mosaic site to vercel, snapshot files are ignored resulting in a 404 when a page is accessed from the site.
+ *
+ * Vercel uses output file tracing (https://nextjs.org/docs/advanced-features/output-file-tracing) to determine what files need to be deployed.
+ * Due to the dynamic nature of mosaic sites, files that are required, the snapshots, are not traced correctly.
+ *
+ * This script adds an entry into the [...route].js.nft.json file for each file in the provided snapshot ensuring that the snapshot files are deployed.
+ *
+ */
+export async function updateTraceFile(config, options) {
+  const { mode, platform } = config.deployment;
+
+  if (mode === 'snapshot-file' && platform === 'vercel') {
+    console.info(
+      '[Mosaic] deployment platform is vercel...adding snapshot files to site output trace'
+    );
+
+    const projectBase = process.cwd();
+    const buildDir = path.posix.join(projectBase, '.next');
+    const snapshotDir = options.out;
+    const snapshotName = options.name ?? new Date().toISOString();
+
+    if (!fsExtra.existsSync(buildDir)) {
+      console.warn(
+        '[Mosaic] unable to update nextjs trace file.  Build the site before generating the snapshot'
+      );
+    }
+    const nftFilePath = path.posix.join(buildDir, 'server', 'pages', '[...route].js.nft.json');
```

Comment:
how did you work this out.. very impressive

Reply:
https://nextjs.org/docs/advanced-features/output-file-tracing

> During next build, Next.js will use [@vercel/nft](https://github.com/vercel/nft) to statically analyze import, require, and fs usage to determine all files that a page might load.

My theory was, the mosaic architecture can't be statically analysed...you need to actually run the app, navigate to a route before the snapshot mdx page is read.  So therefor there will be no snapshot mdx files listed in the trace.